packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -252,7 +264,8 @@ export const BlockSwitcher = ( { clientIds, disabled, isUsingBindings } ) => {
 		clientId: clientIds?.[ 0 ],
 		maximumLength: 35,
 	} );
-	if ( invalidBlocks ) {
+
+	if ( invalidBlocks || selectedHasTemplateLock ) {
```

Comment:
I think this results in no block icon, which doesn't seem like the right thing to do:
<img width="599" alt="Screenshot 2024-09-03 at 2 58 36 PM" src="https://github.com/user-attachments/assets/b02759a5-6256-4ac1-a425-30bbe68e8b88">

Instead there's this block indicator component that's the block icon without a dropdown that could be rendered:
https://github.com/WordPress/gutenberg/blob/84b48f7256dceae44432c8ee3b54433ed0674035/packages/block-editor/src/components/block-switcher/index.js#L266-L283

Reply:
Whoops, I missed that. Good spotting. I'll check out your suggestion. 🙇🏻